prober/websocket.go

```diff
@@ -0,0 +1,137 @@
+// Copyright 2016 The Prometheus Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prober
+
+import (
+	"context"
+	"crypto/tls"
+	"log/slog"
+	"net/http"
+	"net/url"
+
+	"github.com/gorilla/websocket"
+	"github.com/prometheus/blackbox_exporter/config"
+	"github.com/prometheus/client_golang/prometheus"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
+)
+
+func ProbeWebsocket(ctx context.Context, target string, module config.Module, registry *prometheus.Registry, logger *slog.Logger) (success bool) {
+
+	targetURL, err := url.Parse(target)
+	if err != nil {
+		logger.Error("Could not parse target URL", "err", err)
+		return false
+	}
+
+	logger.Debug("probing websocket", "target", targetURL.String())
+
+	httpStatusCode := prometheus.NewGauge(prometheus.GaugeOpts{
+		Name: "probe_http_status_code",
```

Comment:
```suggestion
		Name: "probe_websocket_status_code",
```

`probe_http_status_code` metric name already exists in http prober.  we have probe_<prober>_status_code metric names, so please follow that.